### ISSUE TITLE
range of the inputs when using the pretrained parameters

### ISSUE THREAD

**Issue by sniklaus** [SteffenCzolbe/PerceptualSimilarity#12]
Really interesting work, and thank you for sharing your code!

I would like to train a synthesis network using the Watson-DFT loss and am unsure as to the range that the images should have when using the pretrained parameters. The example in the readme uses zero tensors, which makes it difficult to infer this. I did some digging and found that the data loaders for training normalize the image to a range of -1 to 1 as follows.

https://github.com/SteffenCzolbe/PerceptualSimilarity/blob/727781cddebffeace979ccd32e2667e6cc566850/src/perceptual_sim_training/data/twoafc_dataset.py#L33

As such, should we normalize the (ground truth) image to be in the range of -1 to 1 as well? Thanks for looking into this!

**Comment by SteffenCzolbe** [comment]
Hello @sniklaus,

You found the right piece of code for the 2AFC dataset. The function loads the images, converts them ToTensor (now on a 0..1 scale), then performs this rather lazy normalization step, mapping it to -1..1. The pre-trained functions made available have been trained with this code, thus the correct data input range is -1..1.

Thanks for pointing this out. I notice there might actually be an inconsistency in the repository, where some of the later experiments use data in the 0..1 range instead. I'll investigate further once I get the time.

**Comment by SteffenCzolbe** [comment]
I clarified that data normalization in the Readme (13a9a6530f0f724542b91e72df62b01b56cd55de) and fixed the code to use one consistent normalization (d3646bbfed6d3fd556fe357cd93fe0b22718ba38). I currently have no GPUs available to re-run the affected experiments immediately, but hope I will be able to do so before chistmas.

**Comment by sniklaus** [comment]
Thank you for looking into this so quickly! Feel free to close this issue, unless you want to keep it open until you redid some of the experiments that were inconsistent.

**Comment by sniklaus** [comment]
If the inputs are in the range from -1 to 1, isn't the color conversion wrong since the following RGB to YCbCr conversion expects the input to be in the range of 0 to 1?

https://github.com/SteffenCzolbe/PerceptualSimilarity/blob/13a9a6530f0f724542b91e72df62b01b56cd55de/src/loss/color_wrapper.py#L8

**Comment by SteffenCzolbe** [comment]
Hello,
I've never given the data-normalization much though when building the original repository. We since worked on a more comprehensive fix in the branch 'normalization_fix'. Note that this branch normalizes all data to the 0..1 range instead.

I've re-run all the experiments in the paper with the updated code, and the hyperparameter (beta) choices of the original paper. We observe overall similar artifacts, but observe some over-/under-fitting for some models (see the figures in the project tree of that branch). We are currently re-tuning the hyperparameters, to have properly fit models for all loss functions.

**Comment by sniklaus** [comment]
My apologies for having created such a headache! I wasn't concerned about the results from your paper, the overall story was unlikely to be affected by this oversight. However, kudos to you for being thorough and redoing the experiments!

**Comment by sniklaus** [comment]
Have you by chance had the opportunity to re-tune the hyperparameters by now? I would very much be interested in a new set of weights where the color conversion has been corrected. Thank you for your work!

**Comment by SteffenCzolbe** [comment]
yes, the updated version was published with the neurips proceedings and is also up an arxiv:
https://arxiv.org/abs/2006.15057
